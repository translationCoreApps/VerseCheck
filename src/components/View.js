import React from 'react';
import PropTypes from 'prop-types';
import style from '../css/Style';
// components
import CheckArea from './CheckArea';
import ActionsArea from './ActionsArea';
import SaveArea from './SaveArea';
import DialogComponent from './DialogComponent';
import IconIndicators from './IconIndicators';
import isEqual from 'deep-equal';

class View extends React.Component {

  findIfVerseEdited() {
    const {
      contextIdReducer: {
        contextId
      },
      groupsDataReducer: {
        groupsData
      }
    } = this.props;
    let result = false;

    if (groupsData[contextId.groupId]) {
      let groupData = groupsData[contextId.groupId].filter(groupData => {
        return isEqual(groupData.contextId, contextId);
      });
      result = groupData[0].verseEdits;
    }
    return result;
  }

  render() {
    const {translate} = this.props;
    let titleText;
    let saveArea;
    switch (this.props.mode) {
      case 'edit':
        titleText = translate('edit_verse');
        saveArea = <div />;
        break;
      case 'comment':
        titleText = translate('comment');
        saveArea = <div />;
        break;
      case 'select':
        titleText = translate('select');
        saveArea = <div />;
        break;
      default:
        titleText = translate('step2_check');
        saveArea = <SaveArea {...this.props} />;
    }
    return (
      <div style={style.verseCheck}>
        <div style={{ display: 'flex', flexDirection: 'column', height: '100%' }}>
          <div style={style.verseCheckCard}>
            <div style={style.titleBar}>
              <span>{titleText}</span>
              <IconIndicators
                actions={this.props.actions}
                verseEdited={this.findIfVerseEdited()}
                selectionsReducer={this.props.selectionsReducer}
                verseEditReducer={this.props.verseEditReducer}
                commentsReducer={this.props.commentsReducer}
                remindersReducer={this.props.remindersReducer}
                translate={translate}
              />
            </div>
            <CheckArea {...this.props} />
            <ActionsArea {...this.props} />
          </div>
          {saveArea}
        </div>
        <DialogComponent
          dialogModalVisibility={this.props.dialogModalVisibility}
          handleOpen={this.props.actions.handleOpenDialog}
          handleClose={this.props.actions.handleCloseDialog}
          goToNextOrPrevious={this.props.goToNextOrPrevious}
          skipToNext={this.props.actions.skipToNext}
          skipToPrevious={this.props.actions.skipToPrevious}
          translate={translate}
        />
      </div>
    );
  }
}

View.propTypes = {
  translate: PropTypes.func.isRequired,
  actions: PropTypes.object.isRequired,
  mode: PropTypes.string.isRequired,
  contextIdReducer: PropTypes.shape({
    contextId: PropTypes.object
  }).isRequired,
  groupsDataReducer: PropTypes.shape({
    groupsData: PropTypes.object
  }).isRequired,
  dialogModalVisibility: PropTypes.bool.isRequired,
  goToNextOrPrevious: PropTypes.func,
  selectionsReducer: PropTypes.object.isRequired,
  verseEditReducer: PropTypes.object.isRequired,
  commentsReducer: PropTypes.object.isRequired,
  remindersReducer: PropTypes.object.isRequired
};

export default View;
