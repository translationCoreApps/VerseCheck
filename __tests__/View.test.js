/* eslint-env jest */
import React from 'react';
import fs from 'fs-extra';
import View from '../src/components/View';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import renderer from 'react-test-renderer';

const project = '__tests__/fixtures/project/loadedProjectShortened.json';

describe('View component Tests', () => {
  test('Integrated View test', () => {
    let props = fs.readJSONSync(project);
    props.actions = {
      handleGoToNext: () => jest.fn(),
      handleGoToPrevious: () => jest.fn(),
      handleOpenDialog: () => jest.fn(),
      handleCloseDialog: () => jest.fn(),
      skipToNext: () => jest.fn(),
      skipToPrevious: () => jest.fn(),
      changeSelectionsInLocalState: () => jest.fn(),
      changeMode: () => jest.fn(),
      handleComment: () => jest.fn(),
      checkComment: () => jest.fn(),
      cancelComment: () => jest.fn(),
      saveComment: () => jest.fn(),
      handleTagsCheckbox: () => jest.fn(),
      handleEditVerse: () => jest.fn(),
      checkVerse: () => jest.fn(),
      cancelEditVerse: () => jest.fn(),
      saveEditVerse: () => jest.fn(),
      validateSelections: () => jest.fn(),
      toggleReminder: () => jest.fn(),
      openAlertDialog: () => jest.fn(),
      selectModalTab: () => jest.fn()
    };
    props = {
      ...props,
      cancelSelection: () => jest.fn(),
      clearSelection: () => jest.fn(),
      saveSelection: () => jest.fn(),
      goToNextOrPrevious: () => jest.fn()
    };

    const component = renderer.create(
      <MuiThemeProvider>
        <View {...props} />
      </MuiThemeProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
