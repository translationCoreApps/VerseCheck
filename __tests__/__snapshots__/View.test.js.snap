// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View component Tests before Project Loaded 1`] = `
<div
  style={
    Object {
      "flex": "2 0 280px",
      "margin": "10px",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "borderRadius": "2px",
          "boxShadow": "0 3px 10px var(--background-color)",
          "display": "flex",
          "flex": "auto",
          "flexDirection": "column",
        }
      }
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "var(--accent-color-dark)",
            "color": "var(--reverse-color)",
            "display": "flex",
            "flex": "0 0 40px",
            "fontSize": "16px",
            "fontWeight": "bold",
            "justifyContent": "space-between",
            "padding": "10px",
          }
        }
      >
        <span>
          Select
        </span>
        <div
          style={
            Object {
              "display": "flex",
              "justifyContent": "flex-end",
            }
          }
        >
          <span
            className="glyphicon glyphicon-ok"
            style={
              Object {
                "color": "var(--reverse-color)",
                "margin": "0px 20px",
                "opacity": 0.2,
              }
            }
            title="No selections were found for this check"
          />
          <span
            className="glyphicon glyphicon-pencil"
            style={
              Object {
                "color": "var(--reverse-color)",
                "margin": "0px 20px",
                "opacity": 0.2,
              }
            }
            title="No Verse edits were found for this check"
          />
          <span
            className="glyphicon glyphicon-comment"
            style={
              Object {
                "color": "var(--reverse-color)",
                "margin": "0px 20px",
                "opacity": 0.2,
              }
            }
            title="No comments were found for this check"
          />
          <span
            className="glyphicon glyphicon-bookmark"
            style={
              Object {
                "color": "var(--reverse-color)",
                "margin": "0px 20px",
                "opacity": 0.2,
              }
            }
            title="This check has not been bookmarked for review"
          />
        </div>
      </div>
      <div
        style={
          Object {
            "borderBottom": "1px solid var(--border-color)",
            "display": "flex",
            "flex": "1 0 130px",
            "fontSize": "1.1em",
          }
        }
      >
        <div
          style={
            Object {
              "display": "flex",
              "flex": 1,
              "flexDirection": "column",
            }
          }
        >
          <div
            style={
              Object {
                "display": "flex",
                "flex": "0 0 45px",
                "justifyContent": "space-between",
                "margin": "5px 10px 5px 15px",
              }
            }
          >
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                }
              }
            >
              <span
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                Fwe
              </span>
              <span
                style={
                  Object {
                    "color": "var(--text-color-light)",
                  }
                }
              >
                Titus
                 
                2:5
              </span>
            </div>
          </div>
          <div>
            <div
              style={
                Object {
                  "direction": "ltr",
                  "flex": "auto",
                  "overflowY": "auto",
                  "padding": "0 15px 10px",
                }
              }
            >
              <div
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
              >
                <span>
                  Kono ewe, oambe ziyenderera ne ntaero zisepahara. 
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "borderLeft": "1px solid var(--border-color)",
              "display": "flex",
              "flex": 1,
              "justifyContent": "center",
              "overflowY": "auto",
            }
          }
        >
          <div
            style={
              Object {
                "WebkitUserSelect": "none",
                "alignItems": "center",
                "display": "flex",
                "flex": "1",
                "justifyContent": "center",
                "overflow": "auto",
              }
            }
          >
            <div
              style={
                Object {
                  "overflow": "auto",
                  "padding": "5px",
                  "textAlign": "center",
                }
              }
            >
              <span>
                Please select the translation for:
              </span>
              <br />
              <span>
                <strong
                  style={
                    Object {
                      "color": "var(--accent-color)",
                    }
                  }
                >
                  "God's"
                </strong>
              </span>
              <br />
            </div>
          </div>
        </div>
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "flex": "0 0 55px",
            "justifyContent": "flex-end",
          }
        }
      >
        <button
          className="btn-second"
          onClick={[Function]}
          style={
            Object {
              "alignSelf": "flex-start",
            }
          }
        >
          Cancel
        </button>
        <button
          className="btn-second"
          disabled={true}
          onClick={[Function]}
        >
          <span
            className="glyphicon glyphicon-erase"
            style={
              Object {
                "marginRight": "10px",
              }
            }
          />
          Clear Selection
        </button>
        <button
          className="btn-prime"
          disabled={true}
          onClick={[Function]}
        >
          <span
            className="glyphicon glyphicon-ok"
            style={
              Object {
                "marginRight": "10px",
              }
            }
          />
          Save
        </button>
      </div>
    </div>
    <div />
  </div>
  <div />
</div>
`;
